themes/default/content/blog/iac-recommended-practices-structuring-pulumi-projects/index.md

```diff
@@ -0,0 +1,98 @@
+* **If multiple projects, how many projects?** What's the best way to determine how many projects are needed? What factors play a role in this decision?
+
+## Structuring Pulumi projects
+
+One project, or multiple projects? And if multiple projects, how many projects? These are the core questions the Zephyr needed to resolve. In the end, there are a number of key factors that determine the answers to these questions:
+
+* **Use case:** Zephyr has only a single application (so far!), but in situations where there are multiple applications then keeping those applications (and sometimes their associated infrastructure, depending on other factors) in separate projects is generally recommended.
+* **Company/department/team structure:** How a company/department/team is organized should certainly be taken into account when structuring your Pulumi projects. In Zephyr's early days with Pulumi, a single team managed everything so a single project was fine. Now with multiple teams, Zephyr will likely find using multiple projects to be more effective.
+* **The relationship between resources:** The relationships between resources---such as which resources are dependent upon other resources---is another factor that feeds into your Pulumi project structure. A "shared infrastructure layer" is often best handled in a separate project of its own, but may pull in resources that might otherwise have been kept with an application in its own project. (In some respects, this is a "counter-balance" to the first bullet point.)
+* **Resource lifecycle:** Do resources have a shared lifecycle, meaning they are always created, updated, or deleted together? (Note this is _not_ the same as a dependency on another resource!) If so, then those resources should be placed in the same project.
+* **Change rate of resources:** Some resources may change very infrequently. Consider the example of an AWS VPC and associated objects. Once it is created, it may not change for extended periods of time, if ever. On the other hand, some resources may change frequently (in other words, these resources have a high change rate). In a single project, combining resources that have a high change rate with resources that have a low change rate may be suboptimal.
+
+Now that you have an idea of some of the factors that affect how users structure their Pulumi projects, what are the advantages and disadvantages of each approach?
+
+For single projects, the key advantage is _simplicity._ There's no denying that a single project is simpler and less complex than working with multiple projects. This advantage, though, is also the key disadvantage. A single project does not afford the flexibility to let different teams manage different aspects of the infrastructure, for example, nor does it offer the ability to split resources that don't have a shared lifecycle. You can't run `pulumi destroy` on only _some_ of the resources in a project.
```

Comment:
do we want to make the parallel here to monoliths vs microservices at the application level? its sort of the same concept, its all about the scaling action and the blast radius

Reply:
Yeah, that's a good idea! I'll update the post shortly.

Reply:
I've added this in e569521b

Reply:
In practice, I think you're rarely running `pulumi destroy` on production infrastructure, and while you can't run `destroy` on just some of the infra, you can comment some of it out...

Reply:
@jkodroff OK, that's a fair point. Would you recommend just removing that sentence, or is there a better example to use to help illustrate the point?

Reply:
I think we're really talking about blast radius here, so maybe just say something like "as we mentioned earlier, blast radius is a concern".